lib/ansible/modules/cloud/azure/azure_rm_keyvault_facts.py

```diff
@@ -0,0 +1,193 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2017 Obezimnaka Boms, <t-ozboms@microsoft.com>
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = '''
+---
+module: azure_rm_keyvault_facts
+
+version_added: "2.5"
+
+short_description: Get key vault facts.
+
+description:
+    - Get facts for a specific key vault or all key vaults within a resource group.
+
+options:
+    resource_group:
+        description:
+            - Limit results by resource group. Required when filtering by name.
+    name:
+        description:
+            - Only show results for a specific key vault.
+    tags:
+        description:
+            - Limit results by providing a list of tags. Format tags as 'key' or 'key:value'.
+    top:
+        description:
+            - Maximum number of results to return.
+
+extends_documentation_fragment:
+    - azure
+    - azure_tags
+
+author:
+    - "Obezimnaka Boms @ozboms"
+
+'''
+
+EXAMPLES = '''
+    - name: Get facts for one key vault
+      azure_rm_keyvault_facts:
+        resource_group: Testing
+        name: foobar22
+
+    - name: Get facts for all zones in a resource group
+      azure_rm_keyvault_facts:
+        resource_group: Testing
+        top: 50
+
+    - name: Get facts by tags
+      azure_rm_keyvault_facts:
+        tags:
+          - testing
+'''
+
+RETURN = '''
+azure_keyvault:
+    description: List of key vault dicts.
+    returned: always
+    type: list
+    example:  []
```

Comment:
We should expand on the returned values and at least show the first level keys and what they contain.